llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -208,19 +208,7 @@ static bool sinkScalarOperands(VPlan &Plan) {
     if (NeedsDuplicating) {
       if (ScalarVFOnly)
         continue;
-      VPSingleDefRecipe *Clone;
-      if (auto *SinkCandidateRepR =
-              dyn_cast<VPReplicateRecipe>(SinkCandidate)) {
-        // TODO: Handle converting to uniform recipes as separate transform,
-        // then cloning should be sufficient here.
-        Instruction *I = SinkCandidate->getUnderlyingInstr();
-        Clone = new VPReplicateRecipe(I, SinkCandidate->operands(), true,
-                                      nullptr /*Mask*/, *SinkCandidateRepR);
-        // TODO: add ".cloned" suffix to name of Clone's VPValue.
-      } else {
-        Clone = SinkCandidate->clone();
-      }
-
+      VPSingleDefRecipe *Clone = vputils::getSingleScalarClone(SinkCandidate);
```

Comment:
I wonder what would be missing to completely delegate this narrowing to `narrowToSingleScalar`?

Reply:
The issue is that a replicate GEP has a ScalarIVSteps operand, which is determined dead. I'm trying very hard to get simplifyRecipe to remove the dead ScalarIVSteps.